fiftyone/core/collections.py

```diff
@@ -3632,14 +3634,40 @@ def exclude_fields(self, field_names, _allow_missing=False):
             view = dataset.exclude_fields("predictions.mood")
 
         Args:
-            field_names: a field name or iterable of field names to exclude.
-                May contain ``embedded.field.name`` as well
+            field_names (None): a field name or iterable of field names to
+                exclude. May contain ``embedded.field.name`` as well
+            meta_filter (None): a filter that dynamically excludes fields in
+                the collection's schema according to the specified rule, which
+                can be matched against the field's ``name``, ``type``,
+                ``description``, and/or ``info``. For example:
+
+                -   Use ``meta_filter="2023"`` or
+                    ``meta_filter={"any": "2023"}`` to exclude fields that have
+                    the string "2023" anywhere in their name, type,
+                    description, or info
+                -   Use ``meta_filter={"type": "StringField"}`` or
```

Comment:
FYI I removed `"type": fo.StringField` from the docstring because `ViewStage` kwargs need to be JSON serializable in order to be saved, loaded in the App, etc. 

I didn't touch the actual logic in the `type_filter` though, so the syntax is still available as an undocumented option.

Note: other `ViewStage` classes accept non JSON values, but they are careful to convert the relevant data into JSON serializable values internally in such a way that they can be serialized/deserialized.